test/compojure/api/integration_test.clj

```diff
@@ -49,16 +49,23 @@
   "This middleware appends given value or 1 to a header in request and response."
   ([handler] (middleware* handler 1))
   ([handler value]
-   (fn [request]
-     (let [append #(str % value)
-           request (update-in request [:headers mw*] append)
-           response (handler request)]
-       (update-in response [:headers mw*] append)))))
+   (fn
+     ([request]
+      (let [append #(str % value)
+            request (update-in request [:headers mw*] append)
+            response (handler request)]
+        (update-in response [:headers mw*] append)))
+     ([request respond raise]
+      (let [append #(str % value)
+            request (update-in request [:headers mw*] append)]
+        (handler request #(respond (update-in % [:headers mw*] append)) raise))))))
 
 (defn constant-middleware
   "This middleware rewrites all responses with a constant response."
   [_ res]
-  (constantly res))
+  (fn
```

Comment:
We could have something like `asyncify` etc. function in `compojure.api.async`, which would turn a normal handler/mw into a 1&3 arity version.